cmd/reporting-operator/main.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/jackc/pgx/v4"
+	"github.com/operate-first/curator-operator/internal"
+	"github.com/operate-first/curator-operator/pkg/signals"
+	"github.com/spf13/cobra"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+const (
+	defaultServerPort = ":8080"
+	defaultTimeout    = time.Second * 10
+)
+
+var (
+	reportLog = ctrl.Log.WithName("report")
+)
+
+func newCmd() *cobra.Command {
+	var (
+		serverPort string
+	)
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Starts the http server",
+		Long:  "",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := runReporting(context.Background(), reportLog, serverPort)
+
+			if err != nil {
+				reportLog.Error(err, "error generating report")
+			}
+		},
+	}
+	cmd.Flags().StringVar(&serverPort, "port", defaultServerPort, "configures the default port for the HTTP server")
+
+	return cmd
+}
+
+func main() {
+	opts := zap.Options{
+		Development: true,
+	}
+	logger := zap.New(zap.UseFlagOptions(&opts))
+	ctrl.SetLogger(logger)
+
+	err := newCmd().Execute()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func runReporting(ctx context.Context, logger logr.Logger, httpServerPort string) error {
+	var psqlconn = os.Getenv("POSTGRES_URI")
+	conn, connErr := pgx.Connect(ctx, psqlconn)
+	if connErr != nil {
+		logger.Error(connErr, "failed to open a database connection to postgres")
+		os.Exit(1)
```

Comment:
blocking: return an error here instead of running os.Exit(1) and let the call site handle the error instead of prematurely terminating the program.

not blocking: logger the error and returning an error is somewhat of an anti-pattern.